### PR TITLE
Support help and required argument for the configs

### PR DESCRIPTION
**Pull request by yxdyc**
As the title said, now we can set argument as 
```
cfg.fedopt.optimizer.type = Argument(
        'SGD', description="optimizer type for FedOPT")
```

And the main.py will print help info as follows:
image

### CODE REVIEW

File: federatedscope/attack/auxiliary/attack_trainer_builder.py [hunk 1]
```diff
@@ -3,7 +3,7 @@ def wrap_attacker_trainer(base_trainer, config):
     Args:
         base_trainer (core.trainers.GeneralTorchTrainer): the trainer that
         will be wrapped;
-        config (yacs.config.CfgNode): the configure;
+        config (core.configs.config.CN): the configure;
```
Comment (xieyxclack) [inline federatedscope/attack/auxiliary/attack_trainer_builder.py:6]: `core.configs` -> `federatedscope.core.configs`?

File: federatedscope/autotune/algos.py [hunk 2]
```diff
@@ -68,7 +67,7 @@ def run(self):
 def get_scheduler(init_cfg):
     """To instantiate an scheduler object for conducting HPO
     Arguments:
-        init_cfg (yacs.Node): configuration.
+        init_cfg (core.configs.config.CN): configuration.
```
Comment (xieyxclack) [inline federatedscope/autotune/algos.py:70]: Same as above

File: federatedscope/autotune/algos.py [hunk 3]
```diff
@@ -88,8 +87,8 @@ class Scheduler(object):
     def __init__(self, cfg):
         """
             Arguments:
-                cfg (yacs.Node): dict like object, where each key-value pair
-                corresponds to a field and its choices.
+                cfg (ycore.configs.config.CN): dict like object, where each
```
Comment (xieyxclack) [inline federatedscope/autotune/algos.py:90]: typos

File: federatedscope/core/cmd_args.py [hunk 4]
```diff
@@ -15,12 +16,28 @@ def parse_args(args=None):
...
+        nargs="?",
+        default="all",
+    )
     parser.add_argument('opts',
                         help='See federatedscope/core/configs for all options',
                         default=None,
                         nargs=argparse.REMAINDER)
-    if len(sys.argv) == 1:
+    parse_res = parser.parse_args(args)
+    from federatedscope.core.configs.config import global_cfg
```
Comment (xieyxclack) [inline federatedscope/core/cmd_args.py:29]: Maybe we should move the `import` to the top of the file

File: federatedscope/core/cmd_args.py [hunk 5]
```diff
@@ -15,12 +16,28 @@ def parse_args(args=None):
...
+    init_cfg = global_cfg.clone()
+    if len(sys.argv) == 1 or parse_res.help == "all":
```
Comment (xieyxclack) [inline federatedscope/core/cmd_args.py:31]: We can add some comments for these if sentences

File: materials/notebook/01_quick_start.ipynb [hunk 6]
```diff
@@ -23,7 +23,10 @@
    },
    "source": [
     "## Configurations\n",
-    "**FederatedScope** organize the configuration through `yacs.config.cfgNode` and can be found in `federatedscope.configs`. Please refer to the [official documentation](https://federatedscope.io/refs/index) for specific instructions on how to configure `cfg`."
+    "**FederatedScope** organize the configuration through `core.configs.config
```
Comment (xieyxclack) [inline materials/notebook/01_quick_start.ipynb:26]: a formatting issue?

File: materials/notebook/03_personalized_FL.ipynb [hunk 7]
```diff
@@ -49,7 +49,9 @@
     "For more larger datasets, please refer examples in `scripts/personalization_exp_scripts`.\n",
     "\n",
     "### Configuration\n",
-    "**FederatedScope** organizes the configuration through an extension of `yacs.config.cfgNode`. Please refer to our [official documentation](https://federatedscope.io/docs/own-case/) for specific instructions on how to configure the initial `global_cfg` and customize your own configuration. By default, we provide built-in personalization-related configurations in `federatedscope/core/configs/`."
+    "**FederatedScope** organizes the configuration through an extension of
```
Comment (xieyxclack) [inline materials/notebook/03_personalized_FL.ipynb:52]: Same as above

File: federatedscope/core/configs/yacs_config.py [hunk 8]
```diff
@@ -0,0 +1,600 @@
+# An extended configuration system based on [yacs],
```
Comment (xieyxclack) [inline federatedscope/core/configs/yacs_config.py:1]: We need to add these to `LICENSE` too

**Review by xieyxclack — COMMENTED** [review]
LGTM, please see the inline comments for minor suggestions. THX!

File: federatedscope/core/configs/yacs_config.py (hunk 8 above)
Comment (yxdyc) [reply federatedscope/core/configs/yacs_config.py:1]: > LGTM, please see the inline comments for minor suggestions. THX!

I made modifications according to the comments

File: LICENSE [hunk 9]
```diff
@@ -335,6 +335,24 @@ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 THE SOFTWARE.
 
+------------------------------------------------------------------------------
+
+Code in federatedscope/core/configs/yacs_config.py, the basic code of yacs
+adopts Apache 2.0 License
+
+# Copyright (c) 2018-present, Facebook, Inc.
```
Comment (xieyxclack) [inline LICENSE:343]: uncomment

**Comment by DavdGao** [comment]
@yxdyc  Here is some of my concerns:
- Does it mean that we need to set the arguments by `python federatedscope/main.py --fedopt.optimizer.lr==0.1` in the future? 
- Is it compatible with the old scripts? e.g., `python federatedscope/main.py fedopt.optimizer=0.1`
- In current master branch some namespaces support customized arguments. For example, users can add new arguments under the namespace `train.optimizer`, `finetune.optimizer` and `fedopt.optimizer` (we have set `new_allowed=True`), which haven't been pre-defined in our config files. I'm not sure if the new mechanism still supports it.

**Comment by yxdyc** [comment]
> @yxdyc Here is some of my concerns:
> 
> * Does it mean that we need to set the arguments by `python federatedscope/main.py --fedopt.optimizer.lr==0.1` in the future?
> * Is it compatible with the old scripts? e.g., `python federatedscope/main.py fedopt.optimizer=0.1`
> * In current master branch some namespaces support customized arguments. For example, users can add new arguments under the namespace `train.optimizer`, `finetune.optimizer` and `fedopt.optimizer` (we have set `new_allowed=True`), which haven't been pre-defined in our config files. I'm not sure if the new mechanism still supports it.


Thanks for the questions. I have added de_arguments for [compatibility with historical codes](https://github.com/alibaba/FederatedScope/pull/294/commits/e0430ea417d905b5b947c9c5bf28e07494eff3ab)

**Review by xieyxclack — APPROVED** [review]
LGTM, we can add descriptions of configs later. Thanks!

**Pull request merged**